pkgs/shells/nushell/default.nix

```diff
@@ -56,8 +54,17 @@ rustPlatform.buildRustPackage {
 
   checkPhase = ''
     runHook preCheck
-    echo "Running cargo test"
-    HOME=$(mktemp -d) cargo test
+    (
+      # The skipped tests all fail in the sandbox because in the nushell test playground,
+      # `$HOME` is unset, so nu fallbacks to looking up the passwd dir of the build user
+      # which is /var/empty, while the expectations do respect $HOME.
+      set -x
+      HOME=$(mktemp -d) cargo test -j $NIX_BUILD_CORES --offline -- \
+        --test-threads=$NIX_BUILD_CORES \
+        --skip=repl::test_config_path::test_default_config_path \
+        --skip=repl::test_config_path::test_xdg_config_bad \
+        --skip=repl::test_config_path::test_xdg_config_empty
```

Comment:
Is it possible to do this by just setting `cargoTestFlags` and exporting `HOME` in the `preCheck` hook?

Reply:
i really wanted to write this:
```nix
{
  preCheck = ''
    export HOME=$(mktemp -d)
  '';
  checkType = "debug"; # i also tried the default release here
  checkFlags = [
    "--skip=repl::test_config_path::test_default_config_path"
    "--skip=repl::test_config_path::test_xdg_config_bad"
    "--skip=repl::test_config_path::test_xdg_config_empty"
  ];
}
```

but for whatever reason that has hundreds of failures. as far as i can tell the only difference is a `--target` flag:

from `nix log` output:
```
++cargo test -j 12 --target aarch64-apple-darwin --offline -- --test-threads=12 --skip=repl::test_config_path::test_default_config_path --skip=repl::test_config_path::test_xdg_config_bad --skip=repl::test_config_path::test_xdg_config_empty
```